src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -76,10 +76,20 @@ public partial class Activity : IDisposable
 
         private byte _w3CIdFlags;
 
+        /*
+         * Note:
+         *  DynamicDependency is used here to prevent the linker from removing the struct GetEnumerator on the internal types.
```

Comment:
Looking more at this code, you could also change the Enumerator implementations like the following:

```C#
            public Enumerator<T> GetEnumerator() => new Enumerator<T>(_first);
            IEnumerator<T> IEnumerable<T>.GetEnumerator() => GetEnumerator();
            IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
```

And the `public Enumerator<T> GetEnumerator()` method would be preserved, since now it is being called. This would be more preferrable, since if we were linking an application, preserving ALL public methods on these types is more than necessary.

Reply:
Good call. I updated the code. Should have just done that initially, would have saved myself a lot of time 🤣 Learned a bit about the linker though so not totally wasted 👍